### PR TITLE
WIP: anbox: Init

### PR DESCRIPTION
**Pull request by bkchr**
###### Motivation for this change

Based on the work of @volth and @Mic92.

Currently its missing networking support. I would like to integrate that before merging this pull request.
I'm open for feedback.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by volth** [comment]
There is also @lukeadams' https://github.com/NixOS/nixpkgs/pull/33970

**Comment by bkchr** [comment]
Yeah, I saw that, but as he stopped working on it, I did not mention it.

File: pkgs/applications/virtualization/anbox/default.nix [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      gtest xorg.libpthreadstubs xorg.libXdmcp SDL2_image
+      makeWrapper
+    ];
+
+    preConfigure = ''
+      # Don't build tests
+      truncate -s 0 cmake/FindGMock.cmake
+      truncate -s 0 tests/CMakeLists.txt
+
+      substituteInPlace scripts/gen-emugl-entries.py --replace '/usr/bin/env python2' "${python2}/bin/python"
```
Comment (Mic92) [inline pkgs/applications/virtualization/anbox/default.nix]: `patchShebangs` is the preferred way here.

File: pkgs/applications/virtualization/anbox/default.nix [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+      cat > $out/bin/anbox-application-manager <<EOF
+      #!${bash}/bin/bash
+      ${systemd}/bin/busctl --user call \
+         org.freedesktop.DBus \
+         /org/freedesktop/DBus \
+         org.freedesktop.DBus \
+         StartServiceByName "su" org.anbox 0
+
+      $out/bin/anbox launch --package=org.anbox.appmgr --component=org.anbox.appmgr.AppViewActivity
+      EOF
```
Comment (Mic92) [inline pkgs/applications/virtualization/anbox/default.nix]: You can use makeWrapper both scripts here. It also has a `--run` parameter for the busctl command.

File: pkgs/applications/virtualization/anbox/default.nix [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  anbox = stdenv.mkDerivation {
+    name = "anbox-${version}";
+    inherit src meta;
+
+    nativeBuildInputs = [ cmake pkgconfig ];
+
+    buildInputs = [
+      boost SDL2 mesa_glu glm protobuf dbus-cpp lxc
+      gtest xorg.libpthreadstubs xorg.libXdmcp SDL2_image
+      makeWrapper
```
Comment (Mic92) [inline pkgs/applications/virtualization/anbox/default.nix]: makeWrapper belongs to nativeBuildInputs.

**Comment by jameysharp** [comment]
@bkchr Did you close your PR because you want somebody else to pick up the work, or do you have some reason to believe this shouldn't be packaged? Either way it'd be helpful to record your thoughts for the next person who's interested (possibly me, but no promises).

**Pull request closed without merging**